lib/member_page.rb

```diff
@@ -0,0 +1,95 @@
+require 'scraped_page'
+
+class String
+  def tidy
+    gsub(/[[:space:]]+/, ' ').strip
+  end
+end
+
+class MemberPage < ScrapedPage
+  field :url do
+    url
+  end
+
+  field :image do
+    noko.at_css('.dep_img img/@src').text.tidy
+  end
+
+  field :name do
+    name_and_id.split('-')[0..-2]
+               .reject { |w| w == 'ru' }
+               .map(&:upcase)
+               .join(' ')
+  end
+
+  field :name_ru do
+    noko.at_css('.dep_info h1').text.tidy
+  end
+
+  field :id do
+    name_and_id.split('-').pop
+  end
+
+  field :area do
+    area_and_id[0]
+  end
+
+  field :area_id do
+    area_and_id[1]
+  end
+
+  field :telephone do
+    contact_nodes.text
+                 .scan(/\(?\+?[0-9]+-?[0-9]+\)\s[0-9]+-[0-9]+-[0-9]+/)
```

Comment:
I think this needs a comment to explain what it's trying to match. It doesn't look like the bets way to do this, but it's hard to know at a glance what it's doing.